pkg/apiserver/util.go

```diff
@@ -43,24 +43,20 @@ func restStorageProviders(
 	}
 }
 
-func completeGenericConfig(cfg *genericapiserver.Config) {
+func completeGenericConfig(cfg *genericapiserver.RecommendedConfig) genericapiserver.CompletedConfig {
 	cfg.Serializer = api.Codecs
-	cfg.Complete()
+	completedCfg := cfg.Complete()
 
 	version := version.Get()
 	// Setting this var enables the version resource. We should populate the
 	// fields of the object from above if we wish to have our own output. Or
 	// establish our own version object somewhere else.
 	cfg.Version = &version
```

Comment:
Seems like it might be useful to introduce another field for the version of the API server vs. the version of the machinery.  This is fine for now.